lib/slashdeploy/service.rb

```diff
@@ -51,21 +51,24 @@ def environments(user, repository)
     # Attempts to lock the environment on the repo.
     #
     # environment - An Environment to lock.
-    # message     - An option message.
+    # options     - A hash of options.
+    #               :message - An optional message.
+    #               :force   - Steal the lock if the environment is already locked.
     #
     # Returns a Lock.
-    def lock_environment(user, environment, message = nil)
+    def lock_environment(user, environment, options = {})
       authorize! user, environment.repository
 
       lock = environment.active_lock
 
       if lock
         return if lock.user == user # Already locked, nothing to do.
```

Comment:
if it's already locked, should we let the user know? With physical locks, if I try to lock a lock that's already locked, I get feedback. Raising something like `EnvironmentLockBySelfError` should be safe because the desired state will still be in effect at the end, just with feedback to the user.